src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -129,15 +129,19 @@ public void configurePipeline(MultiOutputPipelineConfigurer configurer) {
   public void initialize(TransformContext context) throws Exception {
     super.initialize(context);
 
-    client = DlpServiceClient.create(getSettings());
     metrics = context.getMetrics();
   }
 
   @Override
   public void prepareRun(StageSubmitterContext context) throws Exception {
     super.prepareRun(context);
 
+    client = DlpServiceClient.create(getSettings());
     String templateName = config.getCustomTemplate();
+    // if templateName is empty, we don't need to validate the template ID
+    if (Strings.isNullOrEmpty(templateName)) {
```

Comment:
If the `customTemplateEnabled` is set to `true`, `templateName` should be provided, `prepareRun` is executed after macros is evaluated so we should check the value of `customTemplateEnabled` to skip validation rather than skipping on empty `templateName`.

Reply:
Done.